### PR TITLE
Feature/aad integration

### PR DESCRIPTION
**Pull request by ThorSchutze**
Created:
- aad.yml gh action - pipeline to create aks groups and users
- create-aksgroups.sh - shell script to create groups using json files as input params
- create-aksusers.sh - shell script to create users using a json file as input param

### CODE REVIEW

File: .github/workflows/aad.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Aad-Integration-CI
+
+# Controls when the workflow will run
+on:
+  push:
+    branches: [ feature/aad-integration ]
```
Comment (tw3lveparsecs) [inline .github/workflows/aad.yml:6]: update branch to main

File: .github/workflows/aad.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Azure Login
+        uses: Azure/login@v1
+        with:
+          creds: ${{ secrets.THOR_AZURE_CREDENTIALS }}
```
Comment (tw3lveparsecs) [inline .github/workflows/aad.yml:22]: update to ${{ secrets.ENTER_SECRET_NAME }}

File: scripts/create-aksusers.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+while getopts k:a: option
+do
+ case "${option}"
+ in
+ # .github/variables/keyvault/keyvault.json
```
Comment (tw3lveparsecs) [inline scripts/create-aksusers.sh:7]: can we clean up unused code that is commented out

**Review by tw3lveparsecs — CHANGES_REQUESTED** [review]

File: scripts/create-aksusers.sh [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+# set keyvault variables
+export vaultname=$(jq -r '.vaultname' ${keyvault})
+export resourcegroup=$(jq -r '.resourcegroup' ${keyvault})
+
+# Grab token and domain name
+AZURE_TOKEN=$(az account get-access-token --resource-type ms-graph --query accessToken --output tsv)
+export AZURE_TENANTDOMAIN=$(curl -s --header "Authorization: Bearer ${AZURE_TOKEN}" --request GET 'https://graph.microsoft.com/v1.0/domains' | jq -r '.value[] | select(.isDefault == true) | {id}[]')
+
+# for row in $(jq -c -r '(.users | .[])' .github/variables/aad/aad.json); do
```
Comment (tw3lveparsecs) [inline scripts/create-aksusers.sh:62]: can this be removed if its not needed?

File: scripts/create-aksusers.sh [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      echo "Failed to create user '${displayName}'"
+      exit 1
+    fi
+    adduser=true
+  fi
+    i=1
+    while [ ! $(az ad user list --upn ${userPrincipalName} --query "[].objectId" -o tsv) ]
+    do
+      ((i++))
+      # echo "Number $i"
```
Comment (tw3lveparsecs) [inline scripts/create-aksusers.sh:111]: can this be removed if its not needed?

**Review by tw3lveparsecs — CHANGES_REQUESTED** [review]
looks good,  still looks like there is some leftover unused code.

File: scripts/create-aksusers.sh (hunk 5 above)
Comment (ThorSchutze) [reply scripts/create-aksusers.sh]: thought I removed that

**Review by tw3lveparsecs — APPROVED** [review]

**Pull request merged**